src/node/rpc/frontend.h

```diff
@@ -543,6 +543,20 @@ namespace ccf
           update_metrics(ctx, metrics);
           return ctx->serialise_response();
         }
+        catch (const nlohmann::json::exception& e)
+        {
+          ctx->set_response_status(HTTP_STATUS_BAD_REQUEST);
+          ctx->set_response_body(e.what());
```

Comment:
I was going to suggest these bodies should all be prefixed so we can more easily tell what 'kind' of error happened. I think this is better achieved by setting the OData error code once this is integrated with #1980? So (assuming we merge that first), can we distinguish JSON parse errors from URL query errors from (...) with separate error codes?

Reply:
Yes, #1980 should be merged first, and then I'll adapt this PR. The problem is that it's not easy to figure out the different types of parse errors in CCF because it's all a bit scattered around and some errors are indistinguishable in terms of url param vs body parse errors.

Reply:
#1980 is in now.

Reply:
Done. I used `InvalidQueryParameterValue` for the `UrlQueryParseError` exception and `InvalidInput` for the rest.